src/FlashBorrower.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "lib/yield-utils-v2/contracts/token/IERC20.sol";
+import "src/IERC3156FlashLender.sol";
+import "src/IERC3156FlashBorrower.sol";
+
+
+contract FlashBorrower is IERC3156FlashBorrower {
+
+    /// @dev Flash loan action states: do something with flash loan if in this state (see onFlashLoan fn)
+    enum Action {NORMAL, OTHER}
+
+    /// @dev Lender is a fixed IERC3156FlashLender contract defined on deployment
+    IERC3156FlashLender lender;
+
+    constructor (IERC3156FlashLender lender_) {
+        lender = lender_;
+    }
+
+
+    /// @notice Initiate a flash loan
+    /// @dev Approve the token to the lender for the total repayment amount, then take flashloan
+    /// @param token Address of the loan currency
+    /// @param amount Amount of token to be loaned 
+    function flashBorrow(address token, uint256 amount) public {
+        bytes memory data = abi.encode(Action.NORMAL);
+
+        uint256 _allowance = IERC20(token).allowance(address(this), address(lender));
+        uint256 _fee = lender.flashFee(token, amount);
+        uint256 _repayment = amount + _fee;
+
+        IERC20(token).approve(address(lender), _allowance + _repayment);
+        lender.flashLoan(this, token, amount, data);
```

Comment:
```suggestion
        lender.flashLoan(address(this), token, amount, data);
```

Reply:
I am unable to use address(this), as flashloan() requires` IERC3156FlashBorrower receiver` as  parameter to be passed, not the address.

![image](https://user-images.githubusercontent.com/22549197/171521905-6e1e950b-fab6-4a59-b327-76b69df0f7e4.png)

![image](https://user-images.githubusercontent.com/22549197/171521948-14884caa-a01a-4f5a-bdb9-4cef5c3f8332.png)